tests/test_sshfsmount.cpp

```diff
@@ -137,6 +138,36 @@ struct SshfsMount : public mp::test::SftpServerTest
         return request_exec;
     }
 
+    // Mock that fails after executing some commands.
+    auto make_exec_that_executes_and_fails(const std::string& fail_cmd, std::string::size_type& remaining,
+                                           std::string& output, bool& invoked_cmd, bool& invoked_fail)
```

Comment:
Now I look at it again… Couldn't an `mp::optional<std::string> fail_cmd` be added to `make_exec_to_check_commands`?

Reply:
Of course :)

Reply:
I meant that this is basically `make_exec_to_check_commands` with empty `commands` now, so this can be dropped?

`make_exec_that_fails_for` is also just a special case of the above now? You could either make them both overloads, or drop completely :)